### PR TITLE
feature/s21_strpbrk

### LINKED ISSUE #29: `s21_strpbrk` - Finds the first character in str1 matching any in str2.
## Description  
Implement the `s21_strpbrk()` function as part of the string+ library. The function should:
1. Locate the first occurrence in `str1` of any character from `str2`
2. Match the behavior of the standard `strpbrk()` (POSIX.1-2017)
3. Return a pointer to the character found, or NULL if no matches exist

## Requirements
- [x] Follow Google C++ Style Guide  
- [x] 100% test coverage (via Check library)  
- [x] No memory leaks  
- [x] Documentation in header file  
- [x] Match standard library behavior exactly

## Test Cases Needed
- Basic character matching
- No matches found
- Empty string input
- Empty character set
- Unicode/ASCII compatibility
- Multiple matches (should return first occurrence)
- NULL pointer handling

## Files to Modify
- `src/s21_strpbrk.c` (implementation)
- `src/s21_string.h` (declaration)
- `tests/test_strpbrk.c` (unit tests)

## Related Functions
- Depends on `s21_strchr()` implementation

### PR DESCRIPTION
**Pull request by DarkDomian**
# Implement s21_strpbrk() function | Closed: #29 

## Changes
✨ Added `s21_strpbrk()` implementation:
- Scans string for first occurrence of any character from accept set
- Returns pointer to matched character or `S21_NULL`
- Fully matches standard library behavior

🧪 Added comprehensive test coverage:
- 5 test cases covering all edge conditions
- Comparison tests against standard `strpbrk()`
- ASCII compatibility checks

📚 Updated documentation:
- Added function description in `s21_string.h`
- Specified return values and edge cases

## Verification

- [x] All tests pass (Check library)  
- [x] 100% line coverage (gcov)  
- [x] Valgrind-clean (no memory leaks)  
- [ ] Clang-format compliant  

## Example Usage
```cpp
char *result = s21_strpbrk("hello world", "aeiou");
// Returns pointer to first 'e'
```

## testing:
```bash
make style-check
make strpbrk.test
```

### CODE REVIEW

**Review by gashturik — APPROVED** [review]
All good

**Comment by ParthenGithub** [comment]
Style-check is arguing:
```
tests/test_strpbrk.c:8:9: style: Variable 'str' can be declared as pointer to const [constVariablePointer]
  char *str = "hello world";
        ^
tests/test_strpbrk.c:9:9: style: Variable 'keys' can be declared as pointer to const [constVariablePointer]
  char *keys = "aeiou";
        ^
tests/test_strpbrk.c:14:9: style: Variable 'str' can be declared as pointer to const [constVariablePointer]
  char *str = "xyz";
        ^
tests/test_strpbrk.c:15:9: style: Variable 'keys' can be declared as pointer to const [constVariablePointer]
  char *keys = "abc";
        ^
tests/test_strpbrk.c:20:9: style: Variable 'str' can be declared as pointer to const [constVariablePointer]
  char *str = "";
        ^
tests/test_strpbrk.c:21:9: style: Variable 'keys' can be declared as pointer to const [constVariablePointer]
  char *keys = "abc";
        ^
tests/test_strpbrk.c:26:9: style: Variable 'str' can be declared as pointer to const [constVariablePointer]
  char *str = "hello";
        ^
tests/test_strpbrk.c:27:9: style: Variable 'keys' can be declared as pointer to const [constVariablePointer]
  char *keys = "";
        ^
tests/test_strpbrk.c:32:9: style: Variable 'str' can be declared as pointer to const [constVariablePointer]
  char *str = "test\0string";
        ^
tests/test_strpbrk.c:33:9: style: Variable 'keys' can be declared as pointer to const [constVariablePointer]
  char *keys = "\0";
```

File: headers/s21_string.h [hunk 1]
```diff
@@ -45,6 +45,16 @@ char *s21_strncpy(char *dest, const char *src, s21_size_t n);
  */
 char *s21_strerror(int errnum);
 s21_size_t s21_strlen(const char *str);
+
+/**
+ * @brief search a string for any of a set of bytes
+ * @return The `s21_strpbrk()` function returns a pointer to the byte in `str1` that matches one of the bytes in `str2`, or
+ * `S21_NULL` if no such byte is found.
+ *
+ * @version 0.8
```
Comment (ParthenGithub) [inline headers/s21_string.h:132]: 0.8? What this means?

**Review by ParthenGithub — CHANGES_REQUESTED** [review]
Clang-format not pass, check the comment below

File: headers/s21_string.h (hunk 1 above)
Comment (DarkDomian) [reply headers/s21_string.h:132]: Does it matter? I just think it want change in future

**Review by Amfir — CHANGES_REQUESTED** [review]
duplicate ‘const’ declaration specifier in the tests/test_strpbrk.c file 8 line)
Otherwise, I like everything.

File: headers/s21_string.h (hunk 1 above)
Comment (Amfir) [reply headers/s21_string.h:132]: In general, it doesn't really matter, but it seems to me that if we specify the version, then it would be better to agree on a common standard for everyone so that specifying the version makes sense at all.

Comment (ParthenGithub) [reply headers/s21_string.h:132]: > Does it matter? I just think it want change in future

It's just... Whty 0.8? Where is 0.7 or 0.9? Why not 0.666?

So many questions :)

**Review by ParthenGithub — CHANGES_REQUESTED** [review]
```c
tests/test_strpbrk.c:8:9: style: Variable 'str' can be declared as pointer to const [constVariablePointer]
  char *str = "hello world";
        ^
tests/test_strpbrk.c:9:9: style: Variable 'keys' can be declared as pointer to const [constVariablePointer]
  char *keys = "aeiou";
        ^
tests/test_strpbrk.c:14:9: style: Variable 'str' can be declared as pointer to const [constVariablePointer]
  char *str = "xyz";
        ^
tests/test_strpbrk.c:15:9: style: Variable 'keys' can be declared as pointer to const [constVariablePointer]
  char *keys = "abc";
        ^
tests/test_strpbrk.c:20:9: style: Variable 'str' can be declared as pointer to const [constVariablePointer]
  char *str = "";
        ^
tests/test_strpbrk.c:21:9: style: Variable 'keys' can be declared as pointer to const [constVariablePointer]
  char *keys = "abc";
        ^
tests/test_strpbrk.c:26:9: style: Variable 'str' can be declared as pointer to const [constVariablePointer]
  char *str = "hello";
        ^
tests/test_strpbrk.c:27:9: style: Variable 'keys' can be declared as pointer to const [constVariablePointer]
  char *keys = "";
        ^
tests/test_strpbrk.c:32:9: style: Variable 'str' can be declared as pointer to const [constVariablePointer]
  char *str = "test\0string";
        ^
tests/test_strpbrk.c:33:9: style: Variable 'keys' can be declared as pointer to const [constVariablePointer]
  char *keys = "\0";
```

**Comment by gashturik** [comment]
Now:
```c
tests/test_strpbrk.c: In function ‘test_strpbrk_basic_fn’:
tests/test_strpbrk.c:8:9: error: duplicate ‘const’ declaration specifier [-Werror=duplicate-decl-specifier]
    8 |   const const char *str = "hello world";
      |         ^~~~~
```

**Review by ParthenGithub — APPROVED** [review]
